src/main/java/com/google/devtools/build/lib/starlarkbuildapi/RunEnvironmentInfoApi.java

```diff
@@ -61,6 +63,15 @@ public interface RunEnvironmentInfoApi extends StructApi {
       structField = true)
   List<String> getInheritedEnvironment();
 
+//   @StarlarkMethod(
```

Comment:
That was intentional: Adding such a method/struct field would give Starlark rules access to this information in the provider. I wasn't sure whether `arguments` is something that other rules should be able to inspect, so I decided not to expose it for now.

Do you see a reason for depending rules to access the value of `run_under_env`? If there is no clear use case, it may be better not to expose it for now. API creep is real and incompatible changes removing functionality are painful.